### PR TITLE
Make task state a choice option for pulp task list.

### LINKED ISSUE #115: Make task state a choice option for pulp task list
`pulp task list --state nonexistent`

### PR DESCRIPTION
**Pull request by thmo**
fixes #115

### CODE REVIEW

File: pulpcore/cli/core/task.py [hunk 1]
```diff
@@ -25,7 +25,14 @@ def task(ctx: click.Context, pulp_ctx: PulpContext) -> None:
                 "name__contains",
                 help=_("List only tasks whose name contains this."),
             ),
-            click.option("--state", help=_("List only tasks in this state.")),
+            click.option(
+                "--state",
+                type=click.Choice(
+                    ["waiting", "skipped", "running", "completed", "failed", "canceled"],
+                    case_sensitive=False,
```
Comment (mdellweg) [inline pulpcore/cli/core/task.py:32]: how does that work?
Is "wAiTiNg" transformed into "waiting" automatically?

Comment (thmo) [reply pulpcore/cli/core/task.py:32]: Yes, as far as I understand (just tested with a minimal example), setting `case_sensitive` to `False` causes the value passed to be normalized to lowercase.

Comment (mdellweg) [reply pulpcore/cli/core/task.py:32]: Can you add too calls to the tests to show this off?
`--state FinisheD` and `--state cannotwork` for example...

File: tests/scripts/pulpcore/test_task.sh [hunk 2]
```diff
@@ -41,5 +41,9 @@ expect_succ test "$(echo "$OUTPUT" | jq -r '.state')" = "completed"
 
 expect_succ pulp task list --name-contains file
 
+expect_fail pulp task list --state=cannotwork
+expect_succ pulp task list --state=COmPLetED
+expect_succ test "$(echo "$OUTPUT" | jq -r '.[0].state')" = "completed"
```
Comment (mdellweg) [inline tests/scripts/pulpcore/test_task.sh:46]: :tada:

**Review by mdellweg — APPROVED** [review]

**Pull request merged**